### DISCUSSION TITLE
Is there a way to change permissions for providers

### DISCUSSION THREAD

**Discussion by GiovanniGobbs** [nextauthjs/next-auth#1040] [Help] [answered]
Hi,
I'm considering using NextAuth within my project, but one of my use cases requires me to change the permissions list for the Facebook login (https://developers.facebook.com/docs/permissions/reference).

Is it possible to specify additional permissions to a provider?

Thanks

**Comment by balazsorban44** [comment, marked as answer]
Permissions are the same as `scope`. You can certainly add any scope to your provider: https://next-auth.js.org/configuration/providers#oauth-provider-options


Ref: https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow/

**Reply by jeffcordova** [reply]
hi, i tried adding scope to the facebook provider but i get redirected to an error page of facebook saying 'Sorry, something went wrong.' I tried, getting the first_name. Please see my code below:

`Providers.Facebook({
	clientId: process.env.FACEBOOK_ID,
	clientSecret: process.env.FACEBOOK_SECRET,
	scope: ['first_name']
})`

I wanted to get the first_name and the picture. Hope you can help me out. Thanks!

**Reply by balazsorban44** [reply]
would need a full reproduction to be able to help. for starters, try adding the email scope back as well. it's part of the default scope list, maybe it's required.

**Reply by jeffcordova** [reply]
Hi, 

Thanks for the quick reply. I just followed the example code here: https://next-auth.js.org/getting-started/example. Then in the ProvidersFacebook, I just inserted scope:"email,first_name", under the clientSecret. 

You may view my test script here: gh repo clone jeffcordova/nextauth-fb

Thank you.

**Reply by JWPapi** [reply]
@jeffcordova I think I’m a bit late, but for further readers:

You need to  use 
```
           authorization : {
                params : {
                    scope : [
                        'email', 'ads_read', 'ads_management', 'public_profile', 'business_management', 'read_insights'
                    ]
                }
            }
```
